builder/vsphere/supervisor/step_import_image.go

```diff
@@ -0,0 +1,364 @@
+		return
+	}
+
+	logger.Info(fmt.Sprintf("Deleting the ContentLibraryItemImportRequest object %s in namespace %s.",
+		s.ImportImageConfig.ImportRequestName, s.Namespace))
+	ctx := context.Background()
+	itemImportReqObj := &imgregv1.ContentLibraryItemImportRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.ImportImageConfig.ImportRequestName,
+			Namespace: s.Namespace,
+		},
+	}
+	if err := s.KubeWatchClient.Delete(ctx, itemImportReqObj); err != nil {
+		logger.Error("error deleting the ContentLibraryItemImportRequest object : %s", err)
+	} else {
+		logger.Info("Successfully deleted the ContentLibraryItemImportRequest object %s in namespace %s.",
+			s.ImportImageConfig.ImportRequestName, s.Namespace)
+	}
+}
+
+func (s *StepImportImage) initStep(state multistep.StateBag, logger *PackerLogger) error {
+	if err := CheckRequiredStates(state,
+		StateKeySupervisorNamespace,
+		StateKeyKubeClient,
+	); err != nil {
+		logger.Error("error checking required states: %s", err)
+		return err
+	}
+
+	var ok bool
+	if s.Namespace, ok = state.Get(StateKeySupervisorNamespace).(string); !ok {
+		return fmt.Errorf("failed to cast %s to type string", StateKeySupervisorNamespace)
+	}
+	if s.KubeWatchClient, ok = state.Get(StateKeyKubeClient).(client.WithWatch); !ok {
+		return fmt.Errorf("failed to cast %s to type client.WithWatch", StateKeyKubeClient)
+	}
+
+	s.SourceURL = s.ImportImageConfig.ImportSourceURL
+	s.SourceSSLCertificate = s.ImportImageConfig.ImportSourceSSLCertificate
+	s.TargetLocationName = s.ImportImageConfig.ImportTargetLocationName
+	if s.ImportImageConfig.ImportTargetImageType != "" {
+		s.TargetItemType = imgregv1.ContentLibraryItemType(strings.ToUpper(s.ImportImageConfig.ImportTargetImageType))
+	}
+
+	// Use the ImageName from create source config as the target item name.
+	s.TargetItemName = s.CreateSourceConfig.ImageName
+	s.KeepInputArtifact = s.CreateSourceConfig.KeepInputArtifact
+
+	state.Put(StateKeyCleanImportedImage, s.ImportImageConfig.CleanImportedImage)
+	state.Put(StateKeyImportedImageName, s.TargetItemName)
+
+	return nil
+}
+
+func (s *StepImportImage) isImportFeatureEnabled(ctx context.Context, logger *PackerLogger) error {
+	importReq := &imgregv1.ContentLibraryItemImportRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    s.Namespace,
+			GenerateName: "import-",
+		},
+	}
+
+	// Use dry run mode to send an image import creation request to API-Server without applying the resource.
+	err := client.NewDryRunClient(s.KubeWatchClient).Create(ctx, importReq)
+	if err != nil && strings.Contains(err.Error(), importOVFFeatureNotEnabledMsg) {
+		logger.Error("image import feature is not enabled")
+		return err
+	}
+
+	return nil
+}
+
+func (s *StepImportImage) isImportSourceValid() error {
+	if strings.HasPrefix(s.ImportImageConfig.ImportSourceURL, "https://") && s.ImportImageConfig.ImportSourceSSLCertificate == "" {
+		return fmt.Errorf("import request source url certificate is empty")
+	}
+
+	return nil
+}
```

Comment:
Move this to the `Prepare()` function instead so that we can detect this error earlier (before any steps are executed)?

Reply:
Sounds good.